packages/web/src/components/navbar.tsx

```diff
@@ -0,0 +1,308 @@
+
+      // FIXME: copy the same language as the currently visible srcbook - should this be less
+      // implicit?
+      language: props.session.language,
+    });
+    openSrcbook(result.path);
+  }
+
+  return (
+    <>
+      {/* FIXME: After closing this modal, `<body>` has `user-select: none;` on it? */}
+      <GenerateSrcbookModal
+        open={showGenSrcbookModal}
+        setOpen={setShowGenSrcbookModal}
+        openSrcbook={openSrcbook}
+      />
+      <ImportSrcbookModal open={showImportSrcbookModal} onOpenChange={setShowImportSrcbookModal} />
+      <DeleteSrcbookModal open={showDelete} onOpenChange={setShowDelete} session={props.session} />
+      <ExportSrcbookModal open={showSave} onOpenChange={setShowSave} session={props.session} />
+
+      <header className="h-12 w-full flex items-center justify-between fixed bg-background z-50 border-b border-border text-sm">
+        <nav className="flex items-center justify-between px-4 flex-1">
+          <div className="flex items-center gap-2">
+            <NavLink to="/">
+              <h1 className="font-mono font-bold flex items-center space-x-[10px] pr-1">
+                <SrcbookLogo size={20} />
+                <span>Srcbooks</span>
+              </h1>
+            </NavLink>
+
+            <span className="select-none">/</span>
+
+            <DropdownMenu>
+              <DropdownMenuTrigger asChild>
+                <Button variant="icon" className="font-normal px-1 active:translate-y-0">
+                  <div className="flex items-center gap-1">
+                    {props.title}
+                    <ChevronDownIcon size={12} />
+                  </div>
+                </Button>
+              </DropdownMenuTrigger>
+              <DropdownMenuContent align="start" className="w-56">
+                {props.srcbooks
+                  .sort((a, b) => b.openedAt - a.openedAt)
+                  .slice(0, 6)
+                  .map((srcbook) => {
+                    if (srcbook.id === props.session.id) {
+                      return null;
+                    }
+                    const titleCell = srcbook.cells.find((cell) => cell.type === 'title') as
+                      | TitleCellType
+                      | undefined;
+                    if (!titleCell) {
+                      return null;
+                    }
+
+                    return (
+                      <DropdownMenuItem
+                        key={srcbook.id}
+                        onClick={() => navigate(`/srcbooks/${srcbook.id}`)}
+                      >
+                        {titleCell.text}
+                      </DropdownMenuItem>
+                    );
+                  })}
+
+                {/* FIXME: how should more than 6 entries be rendered? */}
+                {props.srcbooks.length > 6 ? (
+                  <DropdownMenuItem asChild>
+                    <Link to="/">More...</Link>
+                  </DropdownMenuItem>
+                ) : null}
```

Comment:
Right now, only the 6 most recent entries are rendered in the inline srcbook picker dropdown. It's unclear from the designs if this is the intention or not, but given this probably should be length limited in some capacity to keep it from overflowing the screen, I limited it to 6, and if there's more than 6, rendered a `More...` item that just goes back to the home view.

Reply:
Sounds good to me